### PR TITLE
Setting up environment variables in a return based way instead of set…

### PR DESCRIPTION
**Pull request by larohra**
…ting up state

## Description
Handling ExtesnionUpdateOnFailure by not maintaining state. 
(A sample PR to highlight the 2 different styles of obtaining this feature)
Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/ga/exthandlers.py [hunk 1]
```diff
@@ -484,16 +485,17 @@ def handle_enable(self, ext_handler_i):
...
 
-        ext_handler_i.enable()
+        ext_handler_i.enable(uninstall_failed=uninstall_failed)
 
     @staticmethod
-    def _update_extension_handler(old_ext_handler_i, ext_handler_i):
+    def _update_extension_handler_and_return_if_failed(old_ext_handler_i, ext_handler_i):
 
-        def execute_old_handler_command(func):
+        def execute_old_handler_command_and_return_if_failed(func):
```
Comment (pgombar) [inline azurelinuxagent/ga/exthandlers.py]: Can you add a docstring for the return value and explain that `True` means it failed? Just by looking at it, I initially thought `True` indicated success as it usually does.

**Review by pgombar — APPROVED** [review]
LGTM

File: azurelinuxagent/ga/exthandlers.py (hunk 1 above)
Comment (larohra) [reply azurelinuxagent/ga/exthandlers.py]: Makes sense, I'll modify the doc

Comment (larohra) [reply azurelinuxagent/ga/exthandlers.py]: Changed the function definition to return True on success

**Pull request closed without merging**